dask_sql/physical/rel/logical/limit.py

```diff
@@ -50,17 +51,24 @@ def _apply_limit(self, df: dd.DataFrame, offset: int, end: int) -> dd.DataFrame:
         Unfortunately, Dask does not currently support row selection through `iloc`, so this must be done using a custom partition function.
         However, it is sometimes possible to compute this window using `head` when an `offset` is not specified.
         """
-        if not offset:
-            # We do a (hopefully) very quick check: if the first partition
-            # is already enough, we will just use this
-            first_partition_length = len(df.partitions[0])
-            if first_partition_length >= end:
-                return df.head(end, compute=False)
-
         # compute the size of each partition
         # TODO: compute `cumsum` here when dask#9067 is resolved
         partition_borders = df.map_partitions(lambda x: len(x))
 
+        # if no offset is specified we can use `head` to compute the window
+        if not offset:
+            if dask_config.get("sql.limit.check-partitions"):
```

Comment:
Right now this would be defaulting to true, so that we would trigger a graph computation on all LIMITs; interested in what others think a sensible default here would be